assets/app/scripts/directives/oscFileInput.js

```diff
@@ -8,15 +8,25 @@ angular.module('openshiftConsole')
         model: "=",
         required: "=",
         disabled: "=ngDisabled",
-        helpText: "@?"
+        helpText: "@?",
+        dropArea: "@?"
       },
       templateUrl: 'views/directives/osc-file-input.html',
       link: function(scope, element){
         scope.helpID = _.uniqueId('help-');
         scope.supportsFileUpload = (window.File && window.FileReader && window.FileList && window.Blob);
         scope.uploadError = false;
+        var dropArea = element;
+        if (scope.dropArea) {
+          dropArea = scope.dropArea;
+        }
```

Comment:
This reminds me, when I was trying this yesterday... We don't give you any
indication of what the drop target is, I dropped it outside of the editor
area and it replaced the whole page with the contents of the file.
On Apr 6, 2016 2:26 PM, "Sam Padgett" <notifications@github.com> wrote:

> In assets/app/scripts/directives/oscFileInput.js
> <https://github.com/openshift/origin/pull/8173#discussion_r58756740>:
>
> >        },
> >        templateUrl: 'views/directives/osc-file-input.html',
> >        link: function(scope, element){
> >          scope.helpID = _.uniqueId('help-');
> >          scope.supportsFileUpload = (window.File && window.FileReader && window.FileList && window.Blob);
> >          scope.uploadError = false;
> > +        var dropArea = element;
> > +        if (scope.dropArea) {
> > +          dropArea = scope.dropArea;
> > +        }
>
> var dropArea = scope.dropArea || element;
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8173/files/fe69812b62eb038054b34c55a4709deaacdbaefc#r58756740>
>